services/wallet/gemini.go

```diff
@@ -0,0 +1,72 @@
+package wallet
+
+import (
+	"context"
+	"strings"
+
+	"github.com/brave-intl/bat-go/libs/clients/gemini"
+	appctx "github.com/brave-intl/bat-go/libs/context"
+	"github.com/brave-intl/bat-go/libs/custodian"
+	errorutils "github.com/brave-intl/bat-go/libs/errors"
+)
+
+type geminix struct {
+	docTypes []string
+}
+
+func newGeminix(docTypePrecedence ...string) *geminix {
+	return &geminix{docTypes: docTypePrecedence}
+}
+
+// GetIssuingCountry returns the issuing country for the provided gemini.ValidatedAccount.
+//
+// GetIssuingCountry will primarily try to use the valid documents attached to the account,
+// if no valid documents are associated with the account and the fallback param is true then the account
+// country code will be used.
+//
+// GetIssuingCountry returns an empty string when no accepted document types are associated with the account.
+func (x *geminix) GetIssuingCountry(acc gemini.ValidatedAccount, fallback bool) string {
+	var issuingCountry string
+
+	if fallback {
+		issuingCountry = acc.CountryCode
+	}
+
+	if len(acc.ValidDocuments) > 0 {
+		issuingCountry = countryForDocByPrecedence(x.docTypes, acc.ValidDocuments)
+	}
+
+	return issuingCountry
+}
```

Comment:
Overall, this particular method is OK, although the boolean flag is just OK. I've spent some time thinking about how to not depend on it, and can only think of a couple of possibilities, one of which is to make this method responsible for deciding whether to fall back or not, by passing in the initial two booleans (linking and the equality of linking ids). That way, it could just check those, and branch off based on the result.

On the call site, it would be:

```golang
issuingCountry := service.gemini.GetIssuingCountry(acc, cl.isLinked(), *cl.LinkingID == linkingID)
```

The method itself could be then:

```golang
func (x *geminix) GetIssuingCountry(acc gemini.ValidatedAccount, isLinked, isLinkIDValid bool) string {
	var issuingCountry string

	if isLinked && isLinkIDValid {
		issuingCountry = acc.CountryCode
	}

	if len(acc.ValidDocuments) > 0 {
		issuingCountry = countryForDocByPrecedence(x.docTypes, acc.ValidDocuments)
	}

	return issuingCountry
}
```

The downside then is that we couple the getting country code logic to the decision making logic, which is not great.

So given that, I think, `fallback` is OK, as it allows to leave the decision up to the call site, and leave this code only concerned with country codes.

Reply:
yes, i wrestled with this one for a while and felt the fallback is the most logical otherwise you have to pass multiple booleans or multiple types. Also "fallback" is term being used by other teams so get issuing country with fallback seems pretty clear. 

It should also be a relatively short lived solution, at some point there reason for it should be fixed and we will no longer accept fallback.